GSA-TTS/FAC
# Refactor Search tests + POM

This PR is linked to:
#3485 

This PR will close:
#3574

In this PR:
- Removed files `gen-info`, `initialize-audit`, & `check-access` from `page_objects`.
- Change import back to `testValidGeneralInfo` in `../support/general-info.js`.
- Now storing the POM elements within an `elements` object rather than keeping multiple element identifiers as strings.
- I added a plugin `downloadFile` for cypress and added it as a task in `cypress.config.js`.
- Under support in Cypress directory, in the `commands.js ` file the line `require(cypress-downloadfile) ` was added.
- `Cypress-downloadfile` was added to `package-lock.json `& `package.json`.
- Added `search-tests.cy.js` to e2e, which has four tests 1) basic search test, 2) advance search test, 3) basic search test that return results and downloads summary report, & 4) test reset button and assert fields are empty.

## How to test:

1. Pull this branch, now since we're using a plugin for cypress, we need to make sure to download it locally on our machine or the tests won't work for search. Make sure you're on the backend directory of the project and run `npm i --save-dev cypress-downloadfile`. Once installed, you should see in your IDE a new folder titled downloads appear in the cypress directory, it should appear just under cypress but before e2e.

2. Run docker as you normally would and once docker is running on another tab run this command `docker compose exec web /bin/bash` this will get you into docker shell. After that, run this command `python manage.py materialized_views --create` . (I had to do this since I didn't have materialized view and my search tests would fail but after running command they passed)

3. Now in your terminal, enter your cypress env variables and run npx cypress open

4. Once cypress is open, run the `full-submission.cy.js` tests should pass may take one or two tries for them but once they are completed run the command `python manage.py materialized_views --refresh` and now run `search-tests.cy.js` 

Note: BEFORE running search tests, you'll notice that in your cypress downloads folder you won't have anything in there but AFTER running search tests, a sf-sac-summary-report.xlsx and/or a single-audit-report.pdf will show up, indicating that the download did work.

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
